modules/library/main/src/main/java/org/geotools/api/data/QueryCapabilities.java

```diff
@@ -95,7 +95,7 @@ public boolean isReliableFIDSupported() {
      * identifier.
      */
     public boolean isUseProvidedFIDSupported() {
-        return false;
+        return true;
```

Comment:
this is likely to break existing applications as it inverts the logic

Reply:
It is not clear, because the method does not carry any logic, it only returns false. I have only one option, not to take this parameter into account when adding feature. I hope this goes well for you.

Reply:
This change is not acceptable, not here at least. 

The QueryCapabilities default state indicates what a generic store without any special ability and support can do. Take for example the shapefile data store, it cannot possibly accept user provided identifies, because identifiers are generated from the row number in the shapefile itself... it just cannot be modified.

There are currently a few stores that override this method:
* PropertyDataStore, MemoryDataStore (always supported, they use a string)
* JDBCDataStore, checking if the table has a primary key column 

In your case, you have to build a custom subclass of QueryCapabilities in the WFS store, and return true or false depending on the abilities of the WFS server you're connecting to (it must be a WFS 1.1 server, idgen is only supported in that version).